### ISSUE TITLE
New Feature Request for json support in CORB

### ISSUE THREAD

**Issue by lsaishanker** [marklogic-community/corb2#96]
It will be nice if CORB can provide support to read and process input file in json format.
Thank you.

**Comment by hansenmc** [comment]
This would be easiest to implement with libraries, such as Jackson or GSON. 

Might want to look into providing streaming JSON parser, in order to be able to parse extremely large files.

We would need a means of specifying what to select. XPath? And/or some dot notation substituting “.” for “/“ like JSONPath https://goessner.net/articles/JsonPath/?

**Comment by hansenmc** [comment]
If using dependencies, need to make sure it is user choice to bring implementation(like Jasypyt) or publish as a separate “plugin” jar.

**Comment by bbandlamudi** [comment]
Jaspyt is dependency is done a little differently using reflection to avoid including jasypt.jar at build time. I am not sure if we can something similar to JSON, but I would assume reflection could get tricky as we might need a lot of classes to import. A separate plugin would be more practical I think.

**Comment by hansenmc** [comment]
I was thinking along the lines of setting any of those dependencies as provided or compileOnly. That way, we could build and bundle the classes, but it wouldn’t be included in the distro or the dependency chain.  A user would need to ensure that those jars were on the classpath at runtime if they wanted to use those classes.

**Comment by bbandlamudi** [comment]
I agree - as long as the users who don't use JSON streaming plugin aren't forced to include jaxb or gjson jars, we should be fine.

**Comment by grtjn** [comment]
mlcp and flux come with a shell script that includes arbitrary jars in an ext/ folder. We copied the mlcp script, and edited it to work for corb. Made it easier for us to invoke it from command-line.
